Sources/SourceKit/sourcekitd/CursorInfo.swift

```diff
@@ -99,7 +99,7 @@ extension SwiftLanguageServer {
 
     // FIXME: SourceKit should probably cache this for us.
     if let settings = self.buildSettingsByFile[uri] {
-      skreq[keys.compilerargs] = settings.compilerArguments
+      skreq[keys.compilerargs] = settings.getSwiftCompilerArguments()
```

Comment:
Rather than do this every time, which is easy to forget, let's mutate the settings from the build system before we put them in `self.buildSettingsByFile`.  I think there are two places (one synchronous call to `settings` and then one in the document update settings method.

Reply:
Mutating it would break the equity check against the build server setting in `openDocument` in swift language server
If we think this is the only way to sync the build server working directory setting, maybe we should change `FileBuildSettings`'s init method directly?